omegaconf/dictconfig.py

```diff
@@ -179,10 +178,6 @@ def _validate_set_merge_impl(self, key: Any, value: Any, is_assign: bool) -> Non
         if value == "???":
             return
 
-        if is_assign and isinstance(value, ValueNode) and self._has_element_type():
-            self._check_assign_value_node(key, value)
-            return
-
```

Comment:
While I like removing unneeded functions, it looks like this function is checking something different (and also producing a different error).
can you explain this change more?

Reply:
Validate set for structured config before was checked with its value nodes now, if we are assigning a structured config to a subscripted dict it will raise an error so, this will never be needed.

Reply:
Gotcha.